storage/src/main/java/fk/prof/storage/D42AsyncStorage.java

```diff
@@ -36,15 +36,19 @@ public D42AsyncStorage(AmazonS3 client, String bucket, ExecutorService executorS
     }
 
     @Override
-    public void storeAsync(String path, InputStream content) {
+    public void storeAsync(String path, InputStream content, long length) {
         CompletableFuture.runAsync(() -> {
             try {
-                client.putObject(bucket, path, content, new ObjectMetadata());
+                ObjectMetadata meta = new ObjectMetadata();
+                meta.setContentLength(length);
+                client.putObject(bucket, path, content, meta);
+                // TODO expose metric for size written
```

Comment:
Looked at the vertx.metrics and you can get the underlying metricsRegistry instance. So we can  expose custom metrics.
[http://vertx.io/docs/vertx-dropwizard-metrics/java/#dropwizard-registry](url)